Create connect-onehouse.md (#7324)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/connect-data-platform/connect-onehouse.md

---
title: "Connect Onehouse"

> id: connect-onehouse

Reviewer: ```suggestion
id: connect-onehouse
sidebar_label: "Connect Onehouse"
```

---

> title: "Connect Onehouse"

Reviewer 2: Hey @andywalner and @alberttwong can you add this to the sidebar? See other connectors [here](https://github.com/dbt-labs/docs.getdbt.com/blob/current/website/sidebars.js#L71) Note: they are in alphabetical order.

Author: done.

id: connect-onehouse
description: "Setup instructions for connecting Onehouse to dbt Cloud"

> sidebar_label: "Connect Onehouse"

Reviewer: ```suggestion
```

pagination_next: null
---

> <Constant name="cloud" /> Supports connecting to [Onehouse SQL](https://www.onehouse.ai/) using the Apache Spark Connector with the Thrift method.

Reviewer 2: ```suggestion
<Constant name="cloud" /> supports connecting to [Onehouse SQL](https://www.onehouse.ai/) using the Apache Spark Connector with the Thrift method.
```

**Connect to a Onehouse SQL Cluster with the `dbt-spark` adapter.**

## Requirements

> * For dbt Cloud, ensure your Onehouse SQL endpoint is accessible via external DNS/IP, whitelisting dbt Cloud IPs.

Reviewer 2: ```suggestion
* For <Constant name="cloud" />, ensure your Onehouse SQL endpoint is accessible via external DNS/IP, whitelisting dbt Cloud IPs.
```

| Cluster | Onehouse does not use this field | |
| Connection Timeout | Number of seconds after which to timeout a connection | 10 |
| Connection Retries | Number of times to attempt connecting to cluster before failing | 0 |
| Organization | Onehouse does not use this field | |
| User | Optional. Not enabled by default. | dbt_cloud_user |
| Auth | Optional, supply if using Kerberos. Not enabled by default. | `KERBEROS` |
| Kerberos Service Name | Optional, supply if using Kerberos. Not enabled by default. | `hive` |

<Lightbox src="/img/onehouse/onehouse-dbt.png" width="70%" title="Onehouse configuration"/>

## dbt project

When using dbt, ensure you add necessary configurations to dbt_project.yml for the dbt connector to write data correctly.

| Field | Description | Required | Default  | Recommended |
| ----- | ----------- | -------- | -------- | -------- |
| materialized | Type of table materialization | Yes |   |  table |
| file_format | Open table format to write | Yes |   | hudi   |
| location_root | Location of the database in DFS | Yes |   |    |
| hoodie.table.type | Merge on Read or Copy on Write | No | cow  | mor   |

dbt_project.yml template

> ```

Reviewer 2: ```suggestion

```yml
```

      +materialized: table | incremental
      +file_format: hudi
      +location_root: s3://lakehouse/demolake/dbt_ecomm/
      +tblproperties:
         hoodie.table.type: mor | cow
```

A dbt_project.yml example if using jaffle shop would be

> ```

Reviewer 2: ```suggestion
```sql
```

---
title: "Connect Onehouse"
id: connect-onehouse
description: "Setup instructions for connecting Onehouse to dbt Cloud"
sidebar_label: "Connect Onehouse"
pagination_next: null
---

> <Snippet path="dbt-onehouse" />

Reviewer 2: Removing this unless you have a snippet to add?
```suggestion
```

<Constant name="cloud" /> supports connecting to [Onehouse SQL](https://www.onehouse.ai/) using the Apache Spark Connector with the Thrift method.

> **Connect to a Onehouse SQL Cluster with the `dbt-spark` adapter.**

Reviewer 3: ```suggestion
:::note
Connect to a Onehouse SQL Cluster with the [dbt-spark ](/connect-data-platform/connect-apache-spark)adapter.**
:::
```

Seeing the preview - this will look much better

**Connect to a Onehouse SQL Cluster with the `dbt-spark` adapter.**

## Requirements

* For dbt Cloud, ensure your Onehouse SQL endpoint is accessible via external DNS/IP, whitelisting dbt Cloud IPs.

> * For dbt Core, you can SSH tunnel into the VPC to access the endpoint, without opening up ports.

Reviewer 3: ```suggestion
* For dbt Core, you can SSH tunnel into the VPC to access the endpoint, without opening up ports.
```

Please remove this section because this is a page for setting up on dbt cloud. I recommend having a new page for dbt core setup

Author: @andywalner FYI

Author: removed.

## What works 

> * All dbt features like `dbt clean`, `dbt compile`, `dbt debug`, `dbt seed`, and `dbt run`.

Reviewer 3: ```suggestion
* All dbt Commands, including: `dbt clean`, `dbt compile`, `dbt debug`, `dbt seed`, and `dbt run`.
```

> * dbt materialized type of "table" and "incremental" 

Reviewer 3: ```suggestion
* dbt materializations: `table` and `incremental`
```

> * dbt materialized type of "table" and "incrementals" 

Reviewer 4: change "incrementals" to "incremental"

Author: fixed

* Apache Hudi table types of Merge on Read (MoR) and Copy on Write (CoW). It is recommended to use MoR for mutable workloads.

## Limitations

> * dbt materialized type cannot be "view".

Reviewer 3: ```suggestion
* Views are not supported
```

Reviewer 3: Related to this - dbt defaults to using view if a materialization isn't specified. Do users want to make sure table is the default for on the project level?

Author: yes...  hudi doesn't support spark view so as a result you can't materialize a view.

Reviewer 3: That makes sense --> I think I would move limitations to before the config section so it makes sense together :)

> * `dbt seed` has row / record limits.

Reviewer 3: do you know the row/record limit?

Author: https://github.com/dbt-labs/dbt-adapters/issues/1040.  I put it at no more than 7500.

Reviewer 3: Ah this issue :) Perhaps we should just call that out here?

Author: i'm hoping that dbt fill fix this.

> * `dbt seed` only supports CoW tables.

Reviewer 3: ```suggestion
* `dbt seed` only supports Copy on Write tables.
```
Just updating this to make it more readable

* `dbt seed` only supports CoW tables.

## dbt connection

Fill in the following fields when creating an **Apache Spark** warehouse connection using the Thrift connection method:

| Field | Description | Examples |
| ----- | ----------- | -------- |
| Method | The method for connecting to Spark | Thrift |
| Hostname | The hostname of your Onehouse SQL Cluster endpoint | `yourProject.sparkHost.com` |
| Port | The port to connect to Spark on | 10000 |
| Cluster | Onehouse does not use this field | |
| Connection Timeout | Number of seconds after which to timeout a connection | 10 |
| Connection Retries | Number of times to attempt connecting to cluster before failing | 0 |

> | Organization | Onehouse does not use this field | |

Reviewer 3: I would just remove the fields that Onehouse does not require for the connection (especially since most of them related to Databricks).

Author: I put it there because the dbt cloud UI has those fields.

Reviewer 3: I wish there were a way to make this cleaner 🤔

| User | Optional. Not enabled by default. | dbt_cloud_user |
| Auth | Optional, supply if using Kerberos. Not enabled by default. | `KERBEROS` |
| Kerberos Service Name | Optional, supply if using Kerberos. Not enabled by default. | `hive` |

> 

Reviewer 4: @alberttwong add image here
![onehouse-dbt](https://github.com/user-attachments/assets/d790e734-e8de-4b0c-afc5-87dff224129e)

Author: added

> <Lightbox src="/img/onehouse/onehouse-dbt.png" width="70%" title="Onehouse configuration"/>

Reviewer 3: @alberttwong do you have any concerns about this screenshot showing actual hostname information? Generally we just put a black-box over it

Author: no concern.  it's just an example.

## dbt project

> When using dbt, ensure you add necessary configurations to dbt_project.yml for the dbt connector to write data correctly.

Reviewer 3: I think you should call out that this is needed for the project to default to this. But you can override them on the resources.yml or model config

Author: it's different for different users especially the location_root.

Reviewer 3: I don't understand your response? By putting it in the `dbt_project.yml` you are making this the default setting for the project. I just want to make sure users know that's what they're doing

Reviewer 3: ```suggestion
We recommend that you set default configurations on the dbt_project.yml to ensure that the adapter executes with Onehouse compatible sql
```

This is what I mean. I also cleaned up the language a little since most users should know that they are using dbt if they're on this page and we call connectors, adapters :)

| Field | Description | Required | Default  | Recommended |
| ----- | ----------- | -------- | -------- | -------- |

> | materialized | Type of table materialization | Yes |   |  table |

Reviewer 3: ```suggestion
| materialized | materialization the project/directory will default to | Yes | without input, `view`  |  `table` |
```

> | file_format | Open table format to write | Yes |   | hudi   |

Reviewer 3: ```suggestion
| file_format | table format the project will default to  | Yes |  N/A | hudi   |
```

> | location_root | Location of the database in DFS | Yes |   |    |

Reviewer 3: ```suggestion
| location_root | Location of the database in DFS | Yes | N/A  | <your_database_location_dfs>  |
```

| hoodie.table.type | Merge on Read or Copy on Write | No | cow  | mor   |

> dbt_project.yml template

Reviewer 3: I'm not quite sure why this is necessary since you have one example below?

Author: one was the example of options and the other is a specific example.

dbt_project.yml template
```
      +materialized: table | incremental
      +file_format: hudi

>       +location_root: s3://lakehouse/demolake/dbt_ecomm/

Reviewer 3: ```suggestion
      +location_root: <storage_uri>
```

      +tblproperties:
         hoodie.table.type: mor | cow
```

> An dbt_project.yml example if using jaffle shop would be

Reviewer 4: @alberttwong change "An" to "A"

Author: fixed

```
models:
  jaffle_shop:
    +file_format: hudi
    +location_root: s3://lakehouse/demolake/dbt_ecomm/
    +tblproperties:

>       hoodie.table.type: mor

Reviewer 4: @alberttwong could you include a record key and precombine key in this example?

Author: that is done on the model layer.   not in the connection.

---
title: "Connect Onehouse"
id: connect-onehouse
sidebar_label: "Connect Onehouse"
description: "Setup instructions for connecting Onehouse to dbt Cloud"
pagination_next: null
---

<Constant name="cloud" /> supports connecting to [Onehouse SQL](https://www.onehouse.ai/) using the Apache Spark Connector with the Thrift method.

:::note

> Connect to a Onehouse SQL Cluster with the [dbt-spark ](/connect-data-platform/connect-apache-spark)adapter.**

Reviewer 2: ```suggestion
Connect to a Onehouse SQL Cluster with the [dbt-spark](/docs/cloud/connect-data-platform/connect-apache-spark) adapter.**
```

| Port | The port to connect to Spark on | 10000 |
| Cluster | Onehouse does not use this field | |
| Connection Timeout | Number of seconds after which to timeout a connection | 10 |
| Connection Retries | Number of times to attempt connecting to cluster before failing | 0 |
| Organization | Onehouse does not use this field | |
| User | Optional. Not enabled by default. | dbt_cloud_user |
| Auth | Optional, supply if using Kerberos. Not enabled by default. | `KERBEROS` |
| Kerberos Service Name | Optional, supply if using Kerberos. Not enabled by default. | `hive` |

<Lightbox src="/img/onehouse/onehouse-dbt.png" width="70%" title="Onehouse configuration"/>

## dbt project

We recommend that you set default configurations on the dbt_project.yml to ensure that the adapter executes with Onehouse compatible sql

| Field | Description | Required | Default  | Recommended |
| ----- | ----------- | -------- | -------- | -------- |
| materialized | materialization the project/directory will default to | Yes | without input, `view`  |  `table` |
| file_format | table format the project will default to  | Yes |  N/A | hudi   |

> | location_root | Location of the database in DFS | Yes | N/A  | <your_database_location_dfs>  |

Reviewer 2: ```suggestion
| location_root | Location of the database in DFS | Yes | N/A  | `<your_database_location_dfs>`  |
```
